gitops.py

```diff
@@ -0,0 +1,226 @@
+import time
+from datetime import datetime, timedelta
+import asyncio
+from typing import Any, Optional, Dict
+import httpx
+import yaml
+from decouple import config
+from loguru import logger
+from httpx import BasicAuth
+from pydantic import BaseModel, ValidationError
+
+# These are the credentials passed by the variables of your pipeline to your tasks and into your env
+PORT_CLIENT_ID = config("PORT_CLIENT_ID")
+PORT_CLIENT_SECRET = config("PORT_CLIENT_SECRET")
+BITBUCKET_USERNAME = config("BITBUCKET_USERNAME")
+BITBUCKET_PASSWORD = config("BITBUCKET_PASSWORD")
+BITBUCKET_API_URL = config("BITBUCKET_HOST")
+BITBUCKET_PROJECTS_FILTER = config(
+    "BITBUCKET_PROJECTS_FILTER", cast=lambda v: v.split(",") if v else None, default=[]
+)
+PORT_API_URL = config("PORT_API_URL", default="https://api.getport.io/v1")
+WEBHOOK_SECRET = config("WEBHOOK_SECRET", default="bitbucket_webhook_secret")
```

Comment:
let's remove the webhook variable since we don't need it for gitops

Reply:
Done